composer/utils/dist.py

```diff
@@ -164,6 +165,28 @@ def barrier() -> None:
                        'support.')
 
 
+def monitored_barrier(timeout: Optional[datetime.timedelta] = None) -> None:
+    """Synchronizes all processes.
+
+    This function blocks until all processes reach this function. Unlike `barrier`, `monitored_barrier`
+    times out and raises an error if not all ranks reach this function by `timeout`.
+
+    .. seealso:: :func:`torch.distributed.barrier`
+    """
+    if dist.is_available() and dist.is_initialized():
+        # monitored_barrier requires gloo backend
+        group_gloo = dist.new_group(backend='gloo')
```

Comment:
Instead of creating the group here, could you create the a gloo group as part of `initailize_dist`, and assign it to a global variable?